builtin/providers/aws/resource_aws_ecs_service.go

```diff
@@ -510,6 +510,8 @@ func resourceAwsEcsServiceDelete(d *schema.ResourceData, meta interface{}) error
 func resourceAwsEcsLoadBalancerHash(v interface{}) int {
 	var buf bytes.Buffer
 	m := v.(map[string]interface{})
+
+	buf.WriteString(fmt.Sprintf("%s-", m["target_group_arn"].(string)))
```

Comment:
Hey @KensoDev 

Thanks for the work here. I think a very slightly better way to handle this would be to check if `target_group_arn` is actually set before adding it to the hash

I suggest the following:

```
if m["target_group_arn"] != nil {
		buf.WriteString(fmt.Sprintf("%s-", m["target_group_arn"].(string)))
	}
```

This means that the hash will not change for users who have not got that value set

Thoughts?

Paul